### PR TITLE
dvdpayer: fix GENERAL_SYNCHRONIZE

### PR DESCRIPTION
**Pull request by FernetMenta**
1) fix segfault cause by unprotected refcount
2) add missing handler for GENERAL_SYNCHRONIZE

### CODE REVIEW

**Comment by FernetMenta** [comment]
jenkins build this please

File: xbmc/cores/dvdplayer/DVDPlayerVideo.cpp [hunk 1]
```diff
@@ -356,6 +356,7 @@ void CDVDPlayerVideo::Process()
       continue;
     }
 
+    msgType = pMsg->GetMessageType();
```
Comment (afedchin) [inline xbmc/cores/dvdplayer/DVDPlayerVideo.cpp]: I didn't found usage of msgType.

Comment (FernetMenta) [inline xbmc/cores/dvdplayer/DVDPlayerVideo.cpp]: thanks, debug left over

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by Memphiz** [comment]
jenkins is out of order atm

**Comment by FernetMenta** [comment]
jenkins build this please

**Pull request merged**